spec.bs

```diff
@@ -836,6 +807,7 @@ To <dfn>generate and score bids</dfn> given an [=auction config=] |auctionConfig
     1. Let |compWinner| be the result of running [=generate and score bids=] with |component|,
       |auctionConfig|, |global|, and |settings|.
     1. If |compWinner| is failure, return failure.
+    1. If [=recursively wait until configuration input promises resolve=] given |auctionConfig| returns failure, return failure.

```

Comment:
That would run into trouble with concurrently trying to update leadingBidInfo. It may be reasonable to split [=score and rank a bid=] into two parts, one functional (scoring), and one for updating leadingBidInfo ("ranking"), in which case all the scoring could be parallel, and then ranking be done on collected results...

... but at any rate, I would prefer if that wasn't part of this particular CL.  As is, this line is a correctness fix...

Reply:
I'm happy to make this part of #662 instead of this PR.